pkg/collect/longhorn.go

```diff
@@ -0,0 +1,256 @@
+			return nil, errors.Wrapf(err, "failed to marshal engine %s", engine.Name)
+		}
+		key := filepath.Join(dir, engine.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect engineimages.longhorn.io
+	engineImages, err := client.EngineImages(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list engineimages.longhorn.io")
+	}
+	dir = GetLonghornEngineImagesDirectory(ns)
+	for _, engineImage := range engineImages.Items {
+		b, err := yaml.Marshal(engineImage)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal engineimage %s", engineImage.Name)
+		}
+		key := filepath.Join(dir, engineImage.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect instancemanagers.longhorn.io
+	instanceManagers, err := client.InstanceManagers(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list instancemanagers.longhorn.io")
+	}
+	dir = GetLonghornInstanceManagersDirectory(ns)
+	for _, instanceManager := range instanceManagers.Items {
+		b, err := yaml.Marshal(instanceManager)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal instancemanager %s", instanceManager.Name)
+		}
+		key := filepath.Join(dir, instanceManager.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect backingimagemanagers.longhorn.io
+	backingImageManagers, err := client.BackingImageManagers(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list backingimagemanagers.longhorn.io")
+	}
+	dir = GetLonghornBackingImageManagersDirectory(ns)
+	for _, backingImageManager := range backingImageManagers.Items {
+		b, err := yaml.Marshal(backingImageManager)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal backingimagemanager %s", backingImageManager.Name)
+		}
+		key := filepath.Join(dir, backingImageManager.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect backingimages.longhorn.io
+	backingImages, err := client.BackingImages(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list backingimages.longhorn.io")
+	}
+	dir = GetLonghornBackingImagesDirectory(ns)
+	for _, backingImage := range backingImages.Items {
+		b, err := yaml.Marshal(backingImage)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal backingimage %s", backingImage.Name)
+		}
+		key := filepath.Join(dir, backingImage.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect sharemanagers.longhorn.io
+	shareManagers, err := client.ShareManagers(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list sharemagemanagers.longhorn.io")
+	}
+	dir = GetLonghornShareManagersDirectory(ns)
+	for _, shareManager := range shareManagers.Items {
+		b, err := yaml.Marshal(shareManager)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal sharemanager %s", shareManager.Name)
+		}
+		key := filepath.Join(dir, shareManager.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect settings
```

Comment:
This is a bit different than the others. All settings are in settings.yaml to avoid looking through 50 different files to find a setting.